### PR TITLE
Feature 74 create story

### LINKED ISSUE #74: Create story
As a user, I want to be able to create a story.

### PR DESCRIPTION
**Pull request by veikkahonkanen**
Resolves #74

### CODE REVIEW

File: src/components/screens/home/home-screen.styles.ts [hunk 1]
```diff
@@ -48,5 +76,63 @@ export const styles = createStyles({
...
+
+  previewItem: {
+    paddingTop: theme.spacing(2),
+    paddingBottom: theme.spacing(2),
+    paddingLeft: theme.spacing(2),
+    marginLeft: theme.spacing(2),
+    marginRight: theme.spacing(2),
+    marginBottom: theme.spacing(4),
+    border: "1px solid rgba(255,255,255,0.54)",
+    borderRadius: "5px",
```
Comment (Chi-Chu319) [inline src/components/screens/home/home-screen.styles.ts]:
```suggestion
    borderRadius: 5,
```

**Review by Chi-Chu319 — COMMENTED** [review]
LGTM

File: src/components/screens/home/home-screen.tsx [hunk 2]
```diff
@@ -167,52 +296,141 @@ class HomeScreen extends React.Component<Props, State> {
...
+   */
+  private onImportStoryClick = () => {
+    const { accessToken } = this.props;
+    const { viewIndex } = this.state;
+
+    if (viewIndex === 0 || !accessToken) {
+      this.setState({ viewIndex: 2 });
+      return;
+    }
+    // TODO: implement importing of story from the .xml file
```
Comment (anttileppa) [inline src/components/screens/home/home-screen.tsx:349]: What is a story import XML file?

File: src/components/screens/home/home-screen.tsx [hunk 3]
```diff
@@ -27,6 +30,9 @@ interface Props extends WithStyles<typeof styles> {
 interface State {
   stories: Story[];
   selectedStoryId?: string;
+  viewIndex: number;
```
Comment (anttileppa) [inline src/components/screens/home/home-screen.tsx]: viewIndex sounds a bit volatile, could you explain why this is an index based and not e.g. enumeration?

File: src/components/screens/home/home-screen.tsx [hunk 4]
```diff
@@ -167,52 +296,141 @@ class HomeScreen extends React.Component<Props, State> {
...
+      });
+      this.props.history.push(`editor/${createdStory.id}`);
     }
+    
+  }
+
+  /**
+   * Event handler for return button click
+   */
+  onReturnButtonClick = () => {
```
Comment (anttileppa) [inline src/components/screens/home/home-screen.tsx]: add modifier

File: src/components/screens/home/home-screen.tsx [hunk 5]
```diff
@@ -167,52 +296,141 @@ class HomeScreen extends React.Component<Props, State> {
...
+      storyFile: undefined
+    });
+  }
+
+  /**
+   * Event handler for story name change
+   * 
+   * @param event change event
+   */
+  onStoryNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (anttileppa) [inline src/components/screens/home/home-screen.tsx]: add modifier

File: src/components/screens/home/home-screen.tsx [hunk 6]
```diff
@@ -167,52 +296,141 @@ class HomeScreen extends React.Component<Props, State> {
...
+    const { value } = event.target;
+
+    if (!value) {
+      return;
+    }
+    this.setState({ newStoryName: value });
```
Comment (anttileppa) [inline src/components/screens/home/home-screen.tsx]:
```suggestion

    this.setState({ 
      newStoryName: value
    });
```

File: src/components/screens/home/home-screen.tsx [hunk 7]
```diff
@@ -167,52 +296,141 @@ class HomeScreen extends React.Component<Props, State> {
...
+  }
+
+  /**
+   * Event handler for files drop
+   * 
+   * @param file dropped file
+   */
+  onFilesDropped = (files: File[]) => {
```
Comment (anttileppa) [inline src/components/screens/home/home-screen.tsx]: add modifier

File: src/components/screens/home/home-screen.tsx [hunk 8]
```diff
@@ -167,52 +296,141 @@ class HomeScreen extends React.Component<Props, State> {
...
+          coordinates: { x: 200, y: 400 }
         }
       });
 
-      this.setState({ selectedStoryId: story.id });
+      this.setState({
+        stories: [ ...this.state.stories, createdStory ],
+        selectedStoryId: createdStory.id ?? this.state.selectedStoryId
+      });
+      this.props.history.push(`editor/${createdStory.id}`);
```
Comment (anttileppa) [inline src/components/screens/home/home-screen.tsx:393]:
```suggestion

      this.props.history.push(`editor/${createdStory.id}`);
```

File: src/components/screens/home/home-screen.tsx (hunk 6 above)
Comment (anttileppa) [reply src/components/screens/home/home-screen.tsx]:
```suggestion

    this.setState({ newStoryName: value });
```

File: src/components/screens/home/home-screen.tsx [hunk 9]
```diff
@@ -167,52 +296,141 @@ class HomeScreen extends React.Component<Props, State> {
...
+   * Event handler for files drop
+   * 
+   * @param file dropped file
+   */
+  onFilesDropped = (files: File[]) => {
+    if (!files?.length) {
+      return;
+    }
+
+    this.setState({ storyFile: files[0] });
```
Comment (anttileppa) [inline src/components/screens/home/home-screen.tsx]:
```suggestion
    this.setState({ 
      storyFile: files[0] 
    });
```

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Pull request merged**